packages/firebase_core/example/lib/main.dart

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) Invertase Limited <oss@invertase.io> & Contributors
+// ignore_for_file: public_member_api_docs
+
+import 'dart:async';
+import 'dart:developer';
+import 'package:firebase_core/firebase_core.dart';
+import 'package:flutter/material.dart';
+
+void main() => runApp(const MyApp());
+
+class MyApp extends StatelessWidget {
+  const MyApp({super.key});
+
+  String get name => 'foo';
+
+  // fill app keys if needed, you can take ones from
+  // https://github.com/invertase/flutterfire_desktop/blob/main/packages/firebase_core/firebase_core_desktop/example/lib/main.dart
+  FirebaseOptions get firebaseOptions => const FirebaseOptions(
+        appId: 'xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx',
+        apiKey: 'xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx',
+        projectId: 'react-native-firebase-testing',
+        messagingSenderId: 'xxxxxxxxxxxx',
```

Comment:
You may just copy their keys and IDs.

Reply:
If key is found by dart pub publish the error is reported:
```
Package validation found the following error:
* line 20, column 17 of example/lib/main.dart: Potential leak of Google API Key detected.
     ╷
  20 │         apiKey: 'AIzaSyAgUhHU8wSJgO5MVNy95tMT07NEjzMOfz0',
     │                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
     ╵
  Add a git-ignore style pattern to `false_secrets` in `pubspec.yaml`
  to ignore this. See https://dart.dev/go/false-secrets
```

And now I know that those keys are not actually needed. The core example only shows that the key is stored, but actually sent to servers. So the app is fully functional with those placeholder values.
In functions and auth plugins we will have to decide whether we want the keys in repo. We might also use a fake ones that will only work with locally run emulator (Firebase provide emulators for many of the features).

Reply:
Okay, if the example is fully functional without the actual key and IDs, there's no problem.

Please note that you can also deliberately add the file to the `false_secrets` section in the example's pubspec file to avoid the false positive warning.

https://dart.dev/tools/pub/pubspec#false_secrets